src/arith/rewrite_simplify.cc

```diff
@@ -1761,6 +1761,37 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const AndNode* op) {
 
   TVM_TRY_REWRITE(x == c1 && x != c2, x == c1 && c1 != c2);
   TVM_TRY_REWRITE(x != c2 && x == c1, x == c1 && c1 != c2);
+
```

Comment:
Can you add tests for these. Tests using negative numbers too would be great.

Reply:
Good call, and updated.  There were a couple of the rewrite rules that had incorrect behavior for negative numerators, so those are now fixed.

Reply:
Good call, and updated. There were a couple of the rewrite rules that had incorrect behavior for negative numerators, so those are now fixed.